tools/dbtool/src/main/java/com/horizen/DbToolCommandProcessor.java

```diff
@@ -0,0 +1,249 @@
+        printer.print("Usage:\n" +
+                "\tFrom command line: <program name> <storages data dir path> <command name> [<json data>]\n" +
+                "\tFor interactive mode: <command name> [<json data>]\n" +
+                "\tRead command arguments from file: <command name> -f <path to file with json data>\n" +
+                "Supported commands:\n" +
+                      "\thelp\n" +
+                "\tlastVersionID <arguments>\n" +
+                "\trollback <arguments>\n" +
+                "\tversionsList <arguments>\n" +
+                "\texit\n"
+        );
+    }
+
+    private File getStorageFile(JsonNode json) throws IllegalArgumentException, FileNotFoundException {
+        if (!json.has("storage") || !json.get("storage").isTextual()) {
+            throw new IllegalArgumentException("storage is not specified or is not a string");
+        }
+        String storage_to_rollback = json.get("storage").asText();
+        if (!DbTool.storageNames.contains(storage_to_rollback)) {
+            throw new IllegalArgumentException(storage_to_rollback + " is not a valid storage type.");
+        }
+        log.info(json.toString());
+        File storageFile = new File(dataDirAbsolutePath + "/" + storage_to_rollback);
+
+        if (!storageFile.exists()) {
+            log.error("File on path[" + storageFile + "] doesn't exist");
+            throw new FileNotFoundException("No such file: " + storageFile);
+        }
+        return storageFile;
+    }
+
+    private void printRollbackUsageMsg(String error) {
+        printer.print("Error: " + error);
+        printer.print("Usage:\n" +
+                "\trollback {\"storage\":\"<name>\", \"versionToRollback\":<version>}");
+    }
+
+    private void processRollback(JsonNode json) {
+        if (!json.has("versionToRollback") || !json.get("versionToRollback").isTextual()) {
+            printRollbackUsageMsg("versionToRollback is not specified or is not a string.");
+            return;
+        }
+        String versionToRollback = json.get("versionToRollback").asText();
+        ByteArrayWrapper version = new ByteArrayWrapper(BytesUtils.fromHexString(versionToRollback));
+
+        try {
+            File storageFile = getStorageFile(json);
+            Storage storage = new VersionedLevelDbStorageAdapter(storageFile);
+
+            String storageVersionPre  = BytesUtils.toHexString(storage.lastVersionID().get().data());
+            storage.rollback(version);
+            String storageVersionPost = BytesUtils.toHexString(storage.lastVersionID().get().data());
+
+            ObjectNode resJson = new ObjectMapper().createObjectNode();
+            resJson.put("storage", json.get("storage").asText());
+            resJson.put("versionPrevious", storageVersionPre);
+            resJson.put("versionCurrent", storageVersionPost);
+
+            String res = resJson.toString();
+            printer.print(res);
+            storage.close();
```

Comment:
Shouldn't the storage be closed also when an exception occurs?

Reply:
Done, used AutoCloseable interface for extending Storage class